vault/provider.go

```diff
@@ -93,88 +113,342 @@ func Provider() terraform.ResourceProvider {
 				Description: "The namespace to use. Available only for Vault Enterprise",
 			},
 		},
+		ConfigureFunc:  providerConfigure,
+		DataSourcesMap: dataSourcesMap,
+		ResourcesMap:   resourcesMap,
+	}
+}
 
-		ConfigureFunc: providerConfigure,
+// Description is essentially a DataSource or Resource with some additional metadata
+// that helps with maintaining the Terraform Vault Provider.
+type Description struct {
+	// PathInventory is used for taking an inventory of the supported endpoints in the
+	// Terraform Vault Provider and comparing them to the endpoints noted as available in
+	// Vault's OpenAPI description. A list of Vault's endpoints can be obtained by,
+	// from Vault's home directory, running "$ ./scripts/gen_openapi.sh", and then by
+	// drilling into the paths with "$ cat openapi.json | jq ".paths" | jq 'keys[]'".
+	// Here's a short example of how paths and their path variables should be represented:
+	//		"/transit/keys/{name}/config"
+	//		"/transit/random"
+	//		"/transit/random/{urlbytes}"
+	//		"/transit/sign/{name}/{urlalgorithm}"
+	PathInventory []string
+
+	// Defaults to false, but should be marked true if a resource is enterprise only.
+	EnterpriseOnly bool
+
+	Resource *schema.Resource
+}
 
-		DataSourcesMap: map[string]*schema.Resource{
-			"vault_approle_auth_backend_role_id":   approleAuthBackendRoleIDDataSource(),
-			"vault_kubernetes_auth_backend_config": kubernetesAuthBackendConfigDataSource(),
-			"vault_kubernetes_auth_backend_role":   kubernetesAuthBackendRoleDataSource(),
-			"vault_aws_access_credentials":         awsAccessCredentialsDataSource(),
-			"vault_generic_secret":                 genericSecretDataSource(),
-			"vault_policy_document":                policyDocumentDataSource(),
+// TODO add a tool that lists the path inventory, or even performs a check against Vault so it's super easy
```

Comment:
I think this is outdated as of this PR?

Reply:
Odd, I think you must have been looking at the first commit here rather than the PR as a whole. This is no longer present when viewing the code at the second commit.